### ISSUE TITLE
[Hyrbid local] .cw-settings file change would trigger a rebuild

### ISSUE THREAD

**Issue by yangcao77** [eclipse/codewind#808]
**Codewind version:** master
**OS:** Mac 

**IDE extension version:** hybrid master

**Description:**
a single change to `.cw-settings` file would trigger a project rebuild/update. 
From the filewatcher log, it looks like it's trying to upload all files under the project to the container with `cwctl project sync`. 

logs from filewatcher: 
```
[October 22, 18:24:58.455] [3205.210] Received event [MODIFY] /Users/stephanie/Documents/microclimate/codewind/codewind-workspace/goprojecttest/.cw-settings
[October 22, 18:24:59.463] [3206.218] Batch change summary for 818eb940-f51a-11e9-986c-89c9902c2859 @ 1571783098459: [ >.cw-settings ]
[October 22, 18:24:59.464] [3206.219] Calling cwctl project sync with: { [project] [sync] [-p] [/Users/stephanie/Documents/microclimate/codewind/codewind-workspace/goprojecttest] [-i] [818eb940-f51a-11e9-986c-89c9902c2859] [-t] [118971] }
[October 22, 18:24:59.681] [3206.436] Successfully ran installer command: [project] [sync] [-p] [/Users/stephanie/Documents/microclimate/codewind/codewind-workspace/goprojecttest] [-i] [818eb940-f51a-11e9-986c-89c9902c2859] [-t] [118971] 
[October 22, 18:24:59.683] [3206.438] Output:Uploading to http://127.0.0.1:34003/api/v1/projects/818eb940-f51a-11e9-986c-89c9902c2859/remote-bind/upload
Upload status:200 OK for file: .cw-settings
Upload status:200 OK for file: Dockerfile
Upload status:200 OK for file: Jenkinsfile
Upload status:200 OK for file: README.md
Upload status:200 OK for file: chart/microclimategotemplate/Chart.yaml
Upload status:200 OK for file: chart/microclimategotemplate/templates/deployment.yaml
Upload status:200 OK for file: chart/microclimategotemplate/templates/service.yaml
Upload status:200 OK for file: chart/microclimategotemplate/values.yaml
Upload status:200 OK for file: main.go
[.cw-settings Dockerfile Jenkinsfile README.md chart/microclimategotemplate/Chart.yaml chart/microclimategotemplate/templates/deployment.yaml chart/microclimategotemplate/templates/service.yaml chart/microclimategotemplate/values.yaml main.go]
[.cw-settings Dockerfile Jenkinsfile README.md chart/microclimategotemplate/Chart.yaml chart/microclimategotemplate/templates/deployment.yaml chart/microclimategotemplate/templates/service.yaml chart/microclimategotemplate/values.yaml main.go]
Upload end status:200 OK

```

pfe log
```
[22/10/19 22:24:59 goprojecttest] [INFO] Project 818eb940-f51a-11e9-986c-89c9902c2859 file changed
[22/10/19 22:24:59 goprojecttest] [INFO] cw-settings file changed.
[22/10/19 22:24:59 goprojecttest] [INFO] Extension absolute path is: undefined
[22/10/19 22:24:59 goprojecttest] [INFO] Building container for docker project /codewind-workspace/goprojecttest
[22/10/19 22:24:59 goprojecttest] [INFO] Extension absolute path is: undefined
[22/10/19 22:24:59 goprojecttest] [INFO] The log name for 818eb940-f51a-11e9-986c-89c9902c2859 is cw-818eb940-f51a-11e9-986c-89c9902c2859-b434a12a7f8e734730005a396e64820736822c53
[22/10/19 22:24:59 goprojecttest] [INFO] Container name: cw-goprojecttest-818eb940-f51a-11e9-986c-89c9902c2859
[22/10/19 22:24:59 goprojecttest] [INFO] Log name: cw-818eb940-f51a-11e9-986c-89c9902c2859-b434a12a7f8e734730005a396e64820736822c53
[22/10/19 22:24:59 goprojecttest] [INFO] In K8 environment: false
[22/10/19 22:24:59 goprojecttest] [INFO] Beginning container build and run stage
[22/10/19 22:24:59 goprojecttest] [INFO] Detected other file changes, Codewind will build the project
[22/10/19 22:24:59 goprojecttest] [INFO] File change detected. Project will re-build.
[22/10/19 22:24:59 goprojecttest] [INFO] The context root is not set.
[22/10/19 22:24:59 goprojecttest] [INFO] projectInfo.deploymentRegistry: undefined
[22/10/19 22:24:59 goprojecttest] [INFO] Build state changed for project: 818eb940-f51a-11e9-986c-89c9902c2859 from: success, to: inProgress, with message: buildscripts.buildImage
```

**Steps to reproduce:**
1. create a project
2. modify the .cw-settings file and save

**Comment by tobespc** [comment]
So the way the syncing works is that when we are notified of a file change, any files that have been modified since the last notification will be uploaded.

**Comment by rajivnathan** [comment]
Is the issue that a change to a single file (in this case `.cw-settings`) causes all of the files to be uploaded again?

**Comment by yangcao77** [comment]
I only modified `.cw-settings` file, but all files got uploaded.

**Comment by makandre** [comment]
I would also like to point out that when any file is changed and gets uploaded, it triggers Turbine's [updateProjectForNewChange](https://github.com/eclipse/codewind/blob/master/src/pfe/file-watcher/server/src/controllers/projectEventsController.ts#L49) to run.  But in this pathway we never look at the `ignorePath` that a project type defines, so even files that are meant to not trigger a build would trigger the build.

**Comment by rajivnathan** [comment]
Believe this one's portal? FYI @tobespc

**Comment by jcockbain96** [comment]
I'm seeing this on my latest too, could you assign me @tobespc?

**Comment by jcockbain96** [comment]
@yangcao77 from my testing, I think this is fixed now. 

I added the following debug to check when we're hitting the `/upload` API in PFE. 

![Screenshot 2019-11-06 at 14 10 06](https://user-images.githubusercontent.com/31372187/68305700-082aa000-00a0-11ea-811b-777678d43b85.png)

(https://github.com/eclipse/codewind/blob/master/src/pfe/portal/routes/projects/remoteBind.route.js#L149).

When I create a new project (tested with vscode and direct with cwctl), and then modify it's `.cw-settings` I see only the .cw-settings being uploaded in the pfe logs.

``` 
[06/11/19 14:06:57 /portal/routes/projects/fileChanges.route.js] [INFO] Watch status for projectID 70584fd0-009e-11ea-8139-3b9f391f865f: success
[06/11/19 14:06:57 /portal/routes/projects/remoteBind.route.js] [INFO] Temporary project directory doesn't exist, not syncing any files
PATH OF SYNCCCC .cw-settings
[06/11/19 14:07:21 /portal/routes/projects/remoteBind.route.js] [INFO] Removing locally deleted files from project: dave3, ID: 70584fd0-009e-11ea-8139-3b9f391f865f -
[06/11/19 14:07:21 /portal/routes/projects/remoteBind.route.js] [INFO] project is [object Object] file is .cw-settings projectRoot is /app
[06/11/19 14:07:21 dave3] [INFO] Project 70584fd0-009e-11ea-8139-3b9f391f865f file changed
[06/11/19 14:07:21 dave3] [INFO] cw-settings file changed.
[06/11/19 14:07:22 dave3] [INFO] Update the context root to root
[06/11/19 14:07:22 dave3] [INFO] Only .cw-settings file change detected. Project will not re-build.
[06/11/19 14:07:22 dave3] [INFO] The context root was updated successfully.
[06/11/19 14:07:22 dave3] [INFO] Emitting event
 message: projectSettingsChanged
 data: {
  "operationId": "075bdd8f08d1273041887f6a65962734",
  "projectID": "70584fd0-009e-11ea-8139-3b9f391f865f",
  "contextRoot": "/root",
  "status": "success"
}
```

Are you able to verify?

**Comment by yangcao77** [comment]
verified it's been fixed in the latest master branch

**Comment by yangcao77** [comment]
@james-wallis I just tested on che, I'm still seeing this issue. Looks like your PR only fixed local case. 
```
[06/11/19 20:03:06 gotest] [INFO] The health check endpoint is not set.
[06/11/19 20:03:06 gotest] [INFO] Emitting event 
 message: projectSettingsChanged
 data: {
  "operationId": "ebb27f94df86b924f05174004ffe7e2b",
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "status": "success",
  "ports": {
    "exposedPort": "30886",
    "internalPort": "8000"
  }
}
[06/11/19 20:03:06 gotest] [INFO] https
[06/11/19 20:03:06 gotest] [INFO] Attempting to set the ignored path list for file watching with: 
[06/11/19 20:03:06 gotest] [INFO] The ignored path list for file watching is empty, the default list will be used instead: ["/.project","*/node_modules*","*/.git/*","*/.DS_Store","*/*.swp","*/*.swx","*/4913","/load-test*","*/.dockerignore","*/.gitignore","*/*~","/.settings"]
[06/11/19 20:03:06 gotest] [INFO] The ignored path list for file watching is unchanged: /.project,*/node_modules*,*/.git/*,*/.DS_Store,*/*.swp,*/*.swx,*/4913,/load-test*,*/.dockerignore,*/.gitignore,*/*~,/.settings
[06/11/19 20:03:06 gotest] [INFO] Build state changed for project: 39892bf0-00d0-11ea-a5aa-2f044411325e from: success, to: inProgress, with message: buildscripts.buildImage
[06/11/19 20:03:06 gotest] [INFO] Emitting event 
 message: projectStatusChanged
 data: {
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "buildStatus": "inProgress",
  "detailedBuildStatus": "Creating image"
}
[06/11/19 20:03:06 gotest] [INFO] Build container image
[06/11/19 20:03:06 User.js] [INFO] Check for new log files for project 39892bf0-00d0-11ea-a5aa-2f044411325e
[06/11/19 20:03:06 FileWatcher.js] [INFO] Check for new build log files for project 39892bf0-00d0-11ea-a5aa-2f044411325e
[06/11/19 20:03:06 Turbine] [INFO] Fetching logs for project 39892bf0-00d0-11ea-a5aa-2f044411325e
[06/11/19 20:03:06 gotest] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:06 gotest] [INFO] buildlog path:
 [{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/docker.build.log"]}]
[06/11/19 20:03:06 gotest] [INFO] applog path:
[{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/app.log"]}]
[06/11/19 20:03:10 odopython1] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:10 odopython1] [INFO] Pod name was not found while looking up service information for project.
[06/11/19 20:03:14 gotest] [INFO] Container image build stage complete.
[06/11/19 20:03:14 gotest] [INFO] Build state changed for project: 39892bf0-00d0-11ea-a5aa-2f044411325e from: inProgress, to: inProgress, with message: buildscripts.containerBuildSuccess
[06/11/19 20:03:14 gotest] [INFO] Emitting event 
 message: projectStatusChanged
 data: {
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "buildStatus": "inProgress",
  "detailedBuildStatus": "Container image has been built successfully",
  "appImageLastBuild": "1573070594153"
}
[06/11/19 20:03:14 gotest] [INFO] Pushing image
[06/11/19 20:03:14 gotest] [ERROR] containerBuildAndRun : 1414 | Getting image source signatures

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:164116942aa42751f10ff3432b28829481d37946e15f4f9deb9ddf1a9c703a0d

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:126db7aa380de608ce547f22a5e4e72f8d3de5cf0e627ca205e0cc2cadd405f4

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:2ea751c0f96cbbfb06b41b347898fe2d30d0284574189aa189078c21f4699120

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:9674e3075904468cbf3d9e4ef84114903ae43945006b6e5ac331c663d5837c01

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:7a435d49206fef3d0d7862e527cda5bf2d31a6f01ef73017c6f383545790b6ae

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:831b66a484dc30b193f5d64771e8c01fa7e8fa5c8dab3e5adab5bea21b4223a4

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:0c166e16fb49ffcfa6db855744fd298ebb73d95b4b2041d2c410272844fe079f

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:d5107a7a276a8096aa50fd42e2e27044b3abfeac2782517f446abe450d82ee6d

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:44d7af965eeaf2cf69c92faac00e65f20b849a2ef673af5081f49e665b49bc93

[06/11/19 20:03:15 gotest] [ERROR] containerBuildAndRun : 1414 | Copying blob sha256:9abf87f2a85f60a045ccb7c976c1df2956f32439ae9e9c115b1fd2861f05d1c6

[06/11/19 20:03:17 gotest] [ERROR] containerBuildAndRun : 1414 | Copying config sha256:240649b1fb2f7b604e0d1f6aec6690bb184b2f5902a261f23e65cdf9773cefd7

[06/11/19 20:03:18 gotest] [ERROR] containerBuildAndRun : 1414 | Writing manifest to image destination

[06/11/19 20:03:18 gotest] [ERROR] containerBuildAndRun : 1414 | Storing signatures

[06/11/19 20:03:18 gotest] [INFO] Installing helm chart
[06/11/19 20:03:18 gotest] [INFO] Running helm command: helm upgrade,--install,cw-gotest-39892bf0-00d0-11ea-a5aa,--recreate-pods,--values=/file-watcher/scripts/override-values.yaml,--set,image.repository=docker.io/yangcao77/cw-gotest-39892bf0-00d0-11ea-a5aa,/tmp/gotest/microclimategotemplate
[06/11/19 20:03:19 gotest] [INFO] Build log created: /codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/docker.build.log
[06/11/19 20:03:19 gotest] [ERROR] containerBuildAndRun : 1444 | warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.

[06/11/19 20:03:19 gotest] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:19 gotest] [INFO] Validating project using operationId: 20d683c236985101aaf9e9339a371023
[06/11/19 20:03:19 gotest] [INFO] Checking for required files
[06/11/19 20:03:19 gotest] [INFO] Checking for required file /Dockerfile
[06/11/19 20:03:19 gotest] [INFO] Sending validation result
[06/11/19 20:03:19 gotest] [INFO] Build state changed for project: 39892bf0-00d0-11ea-a5aa-2f044411325e from: inProgress, to: success, with message:  
[06/11/19 20:03:19 gotest] [INFO] Emitting event 
 message: projectStatusChanged
 data: {
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "buildStatus": "success",
  "detailedBuildStatus": " ",
  "lastbuild": 1573070599613
}
[06/11/19 20:03:19 gotest] [INFO] Kicking off validation for project: 39892bf0-00d0-11ea-a5aa-2f044411325e
[06/11/19 20:03:19 gotest] [INFO] Emitting event 
 message: projectValidated
 data: {
  "operationId": "20d683c236985101aaf9e9339a371023",
  "projectType": "docker",
  "location": "/codewind-workspace/gotest",
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "status": "success",
  "results": []
}
[06/11/19 20:03:19 gotest] [INFO] Application state changed for project: 39892bf0-00d0-11ea-a5aa-2f044411325e from: started, to: starting
[06/11/19 20:03:19 gotest] [INFO] Emitting event 
 message: projectStatusChanged
 data: {
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "appStatus": "starting",
  "appErrorStatus": " "
}
[06/11/19 20:03:19 FileWatcher.js] [ERROR] projectChanged: failed (project: {
  "operationId": "9b37326200f0b86a2d07e49746e2564f",
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "status": "failed",
  "ignoredPaths": [
    "/.project",
    "*/node_modules*",
    "*/.git/*",
    "*/.DS_Store",
    "*/*.swp",
    "*/*.swx",
    "*/4913",
    "/load-test*",
    "*/.dockerignore",
    "*/.gitignore",
    "*/*~",
    "/.settings"
  ],
  "isHttps": false,
  "appBaseURL": "http://gotest-codewind-workspace07jess4mwuvcvrig-che-che.9.42.26.107.nip.io"
})
[06/11/19 20:03:19 gotest] [INFO] Emitting event 
 message: projectChanged
 data: {
  "operationId": "9b37326200f0b86a2d07e49746e2564f",
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "status": "failed",
  "ignoredPaths": [
    "/.project",
    "*/node_modules*",
    "*/.git/*",
    "*/.DS_Store",
    "*/*.swp",
    "*/*.swx",
    "*/4913",
    "/load-test*",
    "*/.dockerignore",
    "*/.gitignore",
    "*/*~",
    "/.settings"
  ],
  "isHttps": false,
  "appBaseURL": "http://gotest-codewind-workspace07jess4mwuvcvrig-che-che.9.42.26.107.nip.io"
}
[06/11/19 20:03:20 odopython1] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:20 odopython1] [INFO] Pod name was not found while looking up service information for project.
[06/11/19 20:03:21 gotest] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:23 gotest] [INFO] The pod for helm release cw-gotest-39892bf0-00d0-11ea-a5aa is now up
[06/11/19 20:03:23 gotest] [INFO] Clearing the isApplicationPodUp interval
[06/11/19 20:03:23 gotest] [INFO] The project location for 39892bf0-00d0-11ea-a5aa-2f044411325e is /codewind-workspace/gotest
[06/11/19 20:03:23 gotest] [INFO] The container was started successfully for application /codewind-workspace/gotest
[06/11/19 20:03:23 gotest] [INFO] Found the pod while looking up service information for project.
[06/11/19 20:03:23 gotest] [INFO] Pod name: cw-gotest-39892bf0-00d0-11ea-a5aa-7c5c8fffc5-znkvs
[06/11/19 20:03:23 gotest] [INFO] IP address info: cw-gotest-39892bf0-00d0-11ea-a5aa
[06/11/19 20:03:23 gotest] [INFO] Internal port: 8000
[06/11/19 20:03:23 gotest] [INFO] Exposed port: 30886
[06/11/19 20:03:23 gotest] [INFO] All the exposed pod ports: ["8000"]
[06/11/19 20:03:23 gotest] [INFO] Found container information: {"podName":"cw-gotest-39892bf0-00d0-11ea-a5aa-7c5c8fffc5-znkvs","ip":"","serviceName":"cw-gotest-39892bf0-00d0-11ea-a5aa","exposedPort":"30886","internalPort":"8000","podPorts":["8000"]}
[06/11/19 20:03:23 gotest] [INFO] Streaming application logs on Kubernetes for 39892bf0-00d0-11ea-a5aa-2f044411325e
[06/11/19 20:03:23 gotest] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:23 gotest] [INFO] buildlog path:
 [{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/docker.build.log"]}]
[06/11/19 20:03:23 gotest] [INFO] applog path:
[{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/app.log"]}]
[06/11/19 20:03:23 gotest] [INFO] Emitting event 
 message: projectChanged
 data: {
  "operationId": "9b37326200f0b86a2d07e49746e2564f",
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "status": "success",
  "ignoredPaths": [
    "/.project",
    "*/node_modules*",
    "*/.git/*",
    "*/.DS_Store",
    "*/*.swp",
    "*/*.swx",
    "*/4913",
    "/load-test*",
    "*/.dockerignore",
    "*/.gitignore",
    "*/*~",
    "/.settings"
  ],
  "isHttps": false,
  "ports": {
    "exposedPort": "30886",
    "internalPort": "8000"
  },
  "podName": "cw-gotest-39892bf0-00d0-11ea-a5aa-7c5c8fffc5-znkvs",
  "logs": {
    "build": [
      {
        "origin": "workspace",
        "files": [
          "/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/docker.build.log"
        ]
      }
    ],
    "app": [
      {
        "origin": "workspace",
        "files": [
          "/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/app.log"
        ]
      }
    ]
  }
}
[06/11/19 20:03:23 gotest] [INFO] Emitted event: projectChanged
{"operationId":"9b37326200f0b86a2d07e49746e2564f","projectID":"39892bf0-00d0-11ea-a5aa-2f044411325e","status":"success","ignoredPaths":["/.project","*/node_modules*","*/.git/*","*/.DS_Store","*/*.swp","*/*.swx","*/4913","/load-test*","*/.dockerignore","*/.gitignore","*/*~","/.settings"],"isHttps":false,"ports":{"exposedPort":"30886","internalPort":"8000"},"podName":"cw-gotest-39892bf0-00d0-11ea-a5aa-7c5c8fffc5-znkvs","logs":{"build":[{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/docker.build.log"]}],"app":[{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/app.log"]}]}}
[06/11/19 20:03:23 gotest] [INFO] Container build and helm install stage complete
[06/11/19 20:03:23 User.js] [INFO] Check for new log files for project 39892bf0-00d0-11ea-a5aa-2f044411325e
[06/11/19 20:03:23 FileWatcher.js] [INFO] Check for new app log files for project 39892bf0-00d0-11ea-a5aa-2f044411325e
[06/11/19 20:03:23 gotest] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:23 gotest] [INFO] buildlog path:
 [{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/docker.build.log"]}]
[06/11/19 20:03:23 gotest] [INFO] applog path:
[{"origin":"workspace","files":["/codewind-workspace/.logs/gotest-39892bf0-00d0-11ea-a5aa-2f044411325e/app.log"]}]
[06/11/19 20:03:25 gotest] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:25 gotest] [INFO] Application state changed for project: 39892bf0-00d0-11ea-a5aa-2f044411325e from: starting, to: starting, with message: Pinging http://:8000/health and http://:8000/
[06/11/19 20:03:25 gotest] [INFO] Emitting event 
 message: projectStatusChanged
 data: {
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "appStatus": "starting",
  "appErrorStatus": " ",
  "detailedAppStatus": "Pinging http://:8000/health and http://:8000/"
}
[06/11/19 20:03:26 gotest] [INFO] pingInTransitApplications: Application state error message: Http request is failed due to timeout error
[06/11/19 20:03:27 gotest] [INFO] Extension absolute path is: undefined
[06/11/19 20:03:27 ProjectList.js] [INFO] Project gotest started (project ID: 39892bf0-00d0-11ea-a5aa-2f044411325e)
[06/11/19 20:03:27 gotest] [INFO] Emitting event 
 message: projectStatusChanged
 data: {
  "projectID": "39892bf0-00d0-11ea-a5aa-2f044411325e",
  "appStatus": "started",
  "detailedAppStatus": " "
}

```

**Comment by jcockbain96** [comment]
@yangcao77, was that for me or @james-wallis? I'll see if we can check on che

**Comment by malincoln** [comment]
@rajivnathan pls move this to In progress if it requires an additional fix.  Thanks

**Comment by maysunfaisal** [comment]
This is working on Che with latest devfile:

.cw-settings change triggers the right function, so the turbine calls looks good i think

```
[07/11/19 16:45:02 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace4718s60ik0oi6etd:9191
[07/11/19 16:45:02 server.js] [TRACE] PUT /api/v1/projects/d7346ff0-0179-11ea-98b4-c921122910f4/upload called with 
 req.body { isDirectory: false,
  path: '.cw-settings',
  msg:
   'eJyMizEOwjAMRa8SeeYCZYWBMWLFDKG4TUSwK8cgJMTdUUqhC0hsX/+9B3dk5xBaYaObbUUMYekQEBYvkthIOWQvOqGmaT40UsgWV5Ha049uTYdLP8czLxuzodS3C7nQ++5ZlI4+WBzZrt5uDOvaT9o5XIm9Spcy/eMNpJa+m8gPeAYAAP//c6hRVw==' }
[07/11/19 16:45:02 server.js] [TRACE] responded to PUT /api/v1/projects/d7346ff0-0179-11ea-98b4-c921122910f4/upload with status 200 and body { isDirectory: false,
  path: '.cw-settings',
  msg:
   'eJyMizEOwjAMRa8SeeYCZYWBMWLFDKG4TUSwK8cgJMTdUUqhC0hsX/+9B3dk5xBaYaObbUUMYekQEBYvkthIOWQvOqGmaT40UsgWV5Ha049uTYdLP8czLxuzodS3C7nQ++5ZlI4+WBzZrt5uDOvaT9o5XIm9Spcy/eMNpJa+m8gPeAYAAP//c6hRVw==' }
[07/11/19 16:45:02 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace4718s60ik0oi6etd:9191
[07/11/19 16:45:02 server.js] [TRACE] POST /api/v1/projects/d7346ff0-0179-11ea-98b4-c921122910f4/upload/end called with 
 req.body { fileList:
   [ '.cw-settings',
     '.dockerignore',
     '.gitignore',
     'Dockerfile',
     'Dockerfile-build',
     'Dockerfile-tools',
     'README.md',
     'chart/javamicroprofiletemplate/Chart.yaml',
     'chart/javamicroprofiletemplate/bindings.yaml',
     'chart/javamicroprofiletemplate/templates/basedeployment.yaml',
     'chart/javamicroprofiletemplate/templates/deployment.yaml',
     'chart/javamicroprofiletemplate/templates/hpa.yaml',
     'chart/javamicroprofiletemplate/templates/istio.yaml',
     'chart/javamicroprofiletemplate/templates/service.yaml',
     'chart/javamicroprofiletemplate/values.yaml',
     'pom.xml',
     'src/main/java/application/HealthEndpoint.java',
     'src/main/java/application/rest/JaxrsApplication.java',
     'src/main/java/application/rest/RootEndpoint.java',
     'src/main/java/application/rest/v1/Example.java',
     'src/main/liberty/config/jvm.options',
     'src/main/liberty/config/jvmbx.options',
     'src/main/liberty/config/server.xml',
     'src/main/resources/index.html',
     'src/main/webapp/WEB-INF/beans.xml',
     'src/main/webapp/WEB-INF/ibm-web-ext.xml',
     'src/test/java/it/HealthEndpointIT.java' ],
  modifiedList: [ '.cw-settings' ],
  timeStamp: 1573145058534 }
[07/11/19 16:45:02 /portal/routes/projects/remoteBind.route.js] [INFO] Removing locally deleted files from project: testlib1, ID: d7346ff0-0179-11ea-98b4-c921122910f4 - 
[07/11/19 16:45:02 sharedFunctions.js] [DEBUG] copyProject fromPath: /codewind-workspace/cw-temp/testlib1, toPath: /codewind-workspace/testlib1
[07/11/19 16:45:02 server.js] [TRACE] responded to POST /api/v1/projects/d7346ff0-0179-11ea-98b4-c921122910f4/upload/end with status 200 and body { fileList:
   [ '.cw-settings',
     '.dockerignore',
     '.gitignore',
     'Dockerfile',
     'Dockerfile-build',
     'Dockerfile-tools',
     'README.md',
     'chart/javamicroprofiletemplate/Chart.yaml',
     'chart/javamicroprofiletemplate/bindings.yaml',
     'chart/javamicroprofiletemplate/templates/basedeployment.yaml',
     'chart/javamicroprofiletemplate/templates/deployment.yaml',
     'chart/javamicroprofiletemplate/templates/hpa.yaml',
     'chart/javamicroprofiletemplate/templates/istio.yaml',
     'chart/javamicroprofiletemplate/templates/service.yaml',
     'chart/javamicroprofiletemplate/values.yaml',
     'pom.xml',
     'src/main/java/application/HealthEndpoint.java',
     'src/main/java/application/rest/JaxrsApplication.java',
     'src/main/java/application/rest/RootEndpoint.java',
     'src/main/java/application/rest/v1/Example.java',
     'src/main/liberty/config/jvm.options',
     'src/main/liberty/config/jvmbx.options',
     'src/main/liberty/config/server.xml',
     'src/main/resources/index.html',
     'src/main/webapp/WEB-INF/beans.xml',
     'src/main/webapp/WEB-INF/ibm-web-ext.xml',
     'src/test/java/it/HealthEndpointIT.java' ],
  modifiedList: [ '.cw-settings' ],
  timeStamp: 1573145058534 }
[07/11/19 16:45:02 User.js] [DEBUG] Project d7346ff0-0179-11ea-98b4-c921122910f4 file change list received on 1573145058534. chunk number 1 with total chunk 1 for this timestamp.
[07/11/19 16:45:02 testlib1] [INFO] Project d7346ff0-0179-11ea-98b4-c921122910f4 file changed
[07/11/19 16:45:02 testlib1] [INFO] cw-settings file changed.
[07/11/19 16:45:02 FileWatcher.js] [DEBUG] Received filewatcher module response: 
[07/11/19 16:45:02 FileWatcher.js] [DEBUG] {"statusCode":202}
[07/11/19 16:45:02 testlib1] [TRACE] Project settings invoked for project 
[07/11/19 16:45:02 testlib1] [TRACE] Project settings are {"contextRoot":"","internalPort":"999","healthCheck":"","internalDebugPort":"","isHttps":false,"ignoredPaths":[""],"mavenProfiles":[""],"mavenProperties":[""]}
[07/11/19 16:45:02 testlib1] [INFO] Only .cw-settings file change detected. Project will not re-build.
[07/11/19 16:45:02 testlib1] [INFO] The context root is not set.
[07/11/19 16:45:02 testlib1] [INFO] Found the pod while looking up service information for project.
[07/11/19 16:45:02 testlib1] [INFO] Pod name: cw-testlib1-d7346ff0-0179-11ea-98b4-b87f99c85-ttsgr
[07/11/19 16:45:02 testlib1] [INFO] IP address info: cw-testlib1-d7346ff0-0179-11ea-98b4
[07/11/19 16:45:02 FileWatcher.js] [DEBUG] Portal's event listener: projectSettingsChanged
[07/11/19 16:45:02 FileWatcher.js] [DEBUG] Portal's event listener: {"operationId":"0f809a9542e3d2f8e97a77aed561eb01","projectID":"d7346ff0-0179-11ea-98b4-c921122910f4","ports":{"internalPort":"999"},"status":"failed","error":"The requested application port is not exposed: 999"}
[07/11/19 16:45:02 FileWatcher.js] [ERROR] projectSettingsChanged: failed (project: {
  "operationId": "0f809a9542e3d2f8e97a77aed561eb01",
  "projectID": "d7346ff0-0179-11ea-98b4-c921122910f4",
  "ports": {
    "internalPort": "999"
  },
  "status": "failed",
  "error": "The requested application port is not exposed: 999"
})
[07/11/19 16:45:02 FileWatcher.js] [DEBUG] projectSettingsChanged: failed (project d7346ff0-0179-11ea-98b4-c921122910f4)
[07/11/19 16:45:02 FileWatcher.js] [TRACE] projectSettingsChanged: failed (project: {
  "operationId": "0f809a9542e3d2f8e97a77aed561eb01",
  "projectID": "d7346ff0-0179-11ea-98b4-c921122910f4",
  "ports": {
    "internalPort": "999"
  },
  "status": "failed",
  "error": "The requested application port is not exposed: 999"
})
[07/11/19 16:45:02 testlib1] [INFO] Internal port: 9080
[07/11/19 16:45:02 testlib1] [INFO] Exposed port: 30330
[07/11/19 16:45:02 testlib1] [INFO] All the exposed pod ports: ["9080","9443"]
[07/11/19 16:45:02 testlib1] [INFO] The requested application port is not exposed: 999
[07/11/19 16:45:02 testlib1] [INFO] Emitting event 
 message: projectSettingsChanged
 data: {
  "operationId": "0f809a9542e3d2f8e97a77aed561eb01",
  "projectID": "d7346ff0-0179-11ea-98b4-c921122910f4",
  "ports": {
    "internalPort": "999"
  },
  "status": "failed",
  "error": "The requested application port is not exposed: 999"
}
```

**Comment by yangcao77** [comment]
It still does not work for me with the latest devfile. Any change to `.sw-settings` file triggers a rebuild. 
[pfe.docx](https://github.com/eclipse/codewind/files/3821316/pfe.docx)

**Comment by maysunfaisal** [comment]
Confirmed me and Stephanie have the same PFE image(we have the same image id for the pod) and used the latest dev file. But Stephanie sees the behavior described and I dont.

Different behavior with the same devfile and image. So, cwctt sync logic is behaving differently on different clusters.

**Comment by maysunfaisal** [comment]
We could(emphasizing COULD, dont want to lead ppl astray if its not) be seeing https://github.com/eclipse/codewind/issues/1039#issuecomment-551305508 because of this issue. We should probably keep these two issues close together.

**Comment by jcockbain96** [comment]
I'm not able to recreate on Che with latest dev file. @stalleyj is looking at the sync login now. I'm focussing on #1039 first, then will come back to this one.

**Comment by tobespc** [comment]
moving to verify as unable to recreate here

**Comment by malincoln** [comment]
@elsony to have FW help investigate.  Should maybe move back to In progress?

**Comment by yangcao77** [comment]
Can we move this issue back to in progress?
I can always reproduce it with the latest devfile. But **ONLY after a project has been first time created**. 
@maysunfaisal has the same  PFE and sidecar image. He did not see this issue. 

Step to reproduce:
1. create a project
2. wait for it up and running
3. modify the `.cw-settings` file
(see a build being triggerred at this point)

At the same timestamp `1573662165871 `
Filewatcher was reporting the filechange correctly:
```
[2019-11-13 16:23:48.864] [74420.491] Batch change summary for d323bbc0-0631-11ea-b972-dd912b32d8e8@ 1573662227863: [ >.cw-settings >.cw-settings ]
[2019-11-13 16:23:48.864] [74420.491] Calling cwctl project sync with: [d323bbc0-0631-11ea-b972-dd912b32d8e8] { [ project] [ sync] [ -p] [ /projects/goprojtest] [ -i] [ d323bbc0-0631-11ea-b972-dd912b32d8e8] [ -t] [ 1573662165871] }
```

However, Turbine received the wrong list
```
[13/11/19 16:23:49 server.js] [TRACE] POST /api/v1/projects/d323bbc0-0631-11ea-b972-dd912b32d8e8/upload/end called with 
 req.body { fileList:
   [ '.cw-settings',
     'Dockerfile',
     'README.md',
     'chart/microclimategotemplate/Chart.yaml',
     'chart/microclimategotemplate/templates/deployment.yaml',
     'chart/microclimategotemplate/templates/service.yaml',
     'chart/microclimategotemplate/values.yaml',
     'main.go' ],
  modifiedList:
   [ '.cw-settings',
     'Dockerfile',
     'README.md',
     'chart/microclimategotemplate/Chart.yaml',
     'chart/microclimategotemplate/templates/deployment.yaml',
     'chart/microclimategotemplate/templates/service.yaml',
     'chart/microclimategotemplate/values.yaml',
     'main.go' ],
  timeStamp: 1573662165871 }
```

**Comment by yangcao77** [comment]
This is the `cwctl` version maysun and I were using 
```
bash-4.4$ cat cli_version.properties 
# Build date: 2019-11-12-10:55:16
build_info.url=https://download.eclipse.org/codewind/codewind-installer/master/62
build_info.linux.SHA-1=56bca555f770ffb3a2fdba06482f386f21926176
build_info.macos.SHA-1=9cfcb6ff76fa2d72413aadacfdc20c82f4491ee1
build_info.win.SHA-1=6eeef25c1f52814c84f91c0de6839174029a1902
```
It does look like a  cwctl issue, and looks like cwctl behaves differently on different environments.  
I can provide my cluster to help debug.

**Comment by GeekArthur** [comment]
I reinstall Che 7.3.1 (we have to update Che version since Theia editor has been updated to 7.3.1) and use devfile 0.6.0, no rebuild issue and work as expected.

**Comment by elsony** [comment]
Moving back to in progress since @yangcao77 can still reproduce the problem. I have downgraded this to hot since it looks like this problem is specific to certain machines.

**Comment by yangcao77** [comment]
I'm always able to reproduce this issue on both 7.2.0 and 7.3.1 with the latest devfile. 
I can share my cluster if you need an environment to reproduce and debug on this issue.

**Comment by tobespc** [comment]
the double build fix has now gone into 0.6.0, should retest this now

**Comment by tobespc** [comment]
I can't recreate this , can anyone else on here ?

**Comment by yangcao77** [comment]
I can still reproduce this with the latest master devfile on che 7.3.1. 
This is my cluster: http://che-che731.9.42.26.107.nip.io

**Comment by maysunfaisal** [comment]
Yes, I concur with @yangcao77 i.e.; `cwctl` is behaving differently on different environments for the same use-case. From Stephanie's cluster on `.cw-settings` edit:

### FW log
```
[2019-11-26 23:20:58.987] [866.108] Calling cwctl project sync with: [2c5df450-10a3-11ea-b04c-b9c7c95ab668] { [ project] [ sync] [ -p] [ /projects/gotest123] [ -i] [ 2c5df450-10a3-11ea-b04c-b9c7c95ab668] [ -t] [ 1574810360292] }
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
127.0.0.1 - - [26/Nov/2019:23:20:59 +0000] "POST /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload/end HTTP/1.1" 200 2 "-" "Go-http-client/1.1" "-"
[2019-11-26 23:20:59.368] [866.489] Cwctl call completed, elapsed time of cwctl call: 381
[2019-11-26 23:20:59.368] [866.489] Successfully ran installer command: [ project] [ sync] [ -p] [ /projects/gotest123] [ -i] [ 2c5df450-10a3-11ea-b04c-b9c7c95ab668] [ -t] [ 1574810360292] 
[2019-11-26 23:20:59.368] [866.489] Output:time="2019-11-26T23:20:59Z" level=info msg="Status: 200 OK"
```

### PFE log
```
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: '.cw-settings',
  msg:
   'eJxsz79KBDEQx/FXWVIGTRBttNXC8vBKI0dMxk28OLNkRjwQfXaJt8r+ufbzhV8y6tNh1zkVCAUO8kAkTt10Tvnn4NTZMWYUqOjLhupY/1MCXyTdJgj7Rcl8LzJw0xdfGP64R6oQN17Sb3ts3II1Q6VXCDIuNNMWKcLujeJ7AdazYvosdkF3291WqMJMteGPYSWHmVxdX1xOwBby8VyAZfFApLCHerxh9ZsTrL+nq4ZBJGPPTjV7cvilfgIAAP//sBpv0g==' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: '.cw-settings',
  msg:
   'eJxsz79KBDEQx/FXWVIGTRBttNXC8vBKI0dMxk28OLNkRjwQfXaJt8r+ufbzhV8y6tNh1zkVCAUO8kAkTt10Tvnn4NTZMWYUqOjLhupY/1MCXyTdJgj7Rcl8LzJw0xdfGP64R6oQN17Sb3ts3II1Q6VXCDIuNNMWKcLujeJ7AdazYvosdkF3291WqMJMteGPYSWHmVxdX1xOwBby8VyAZfFApLCHerxh9ZsTrL+nq4ZBJGPPTjV7cvilfgIAAP//sBpv0g==' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: 'Dockerfile',
  msg:
   'eJw0i8EKwjAUBH9lyd02R/EmpoJIjUREwefhSSUE003R+v/SgsfdmTHb4FvEkpVxlXV8fkYIw/mA/tWlN2odBgjXzqGaRw3hxYe924U/nOxY8Pim3GFR0GsiKgg3rcNNzJxNp5i7sLke/anB0lorNL8AAAD//0d1J9Q=' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: 'Dockerfile',
  msg:
   'eJw0i8EKwjAUBH9lyd02R/EmpoJIjUREwefhSSUE003R+v/SgsfdmTHb4FvEkpVxlXV8fkYIw/mA/tWlN2odBgjXzqGaRw3hxYe924U/nOxY8Pim3GFR0GsiKgg3rcNNzJxNp5i7sLke/anB0lorNL8AAAD//0d1J9Q=' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: 'README.md',
  msg: 'eJxSUlZIzw9JzS3ISSxJVQIEAAD//x+TBJo=' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: 'README.md',
  msg: 'eJxSUlZIzw9JzS3ISSxJVQIEAAD//x+TBJo=' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/Chart.yaml',
  msg:
   'eJw8zLEKQjEMRuFXCX2A0q53cxPcnbLE668WbtOQxj6/IOh4zvAlsXaFzzZ0o1VZxezfnGounFjvmLs3i+890RlHp/0lHvQYTpf3Da4ITFaVjo2eI9DtkADr+mkl11zSJwAA///s4yfK' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/Chart.yaml',
  msg:
   'eJw8zLEKQjEMRuFXCX2A0q53cxPcnbLE668WbtOQxj6/IOh4zvAlsXaFzzZ0o1VZxezfnGounFjvmLs3i+890RlHp/0lHvQYTpf3Da4ITFaVjo2eI9DtkADr+mkl11zSJwAA///s4yfK' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/templates/deployment.yaml',
  msg:
   'eJzEUj2P2zAM/SuElhuK5HCr1ivQrTWK4iYBBSMTF6EyJUi00eCq/17I9vnjkgydyiUByfce35MVRvdCKbvAGui3ENe/+XF4OpHgk+FfjlsNnyn6cOmIxXBHgi0KasMAjB1pMOrtDY7PZ0xy/IodQSmHdoWouunxRD6PIABbVzU8XMPWznwW/IFE0aMlMOqTUWDUT6OglAfDOZIdGeuGs5g1VPwL+p7ycW4+h54FSql7Ql30KDSfsbNSa3dkLYzxtr1MnqyENJmbangP0ijbp7RYB1gOHc0HFnRMaRU63A9yK+A6fF22Zpdjr3oN2UlIFyhFX40FX29RNb33TfDOXnbBTZi4DOfwpoohyTahw+qnCfVRN0SZ0uAsvf/W+Y7Lu4GYcm5SONGGE+AsEr+Q7HoAEeWs4fFD819VE2Hr/odsDn2ytA2PeNhFOX0FzbfvP7ZiQxX48Ox31JRhw+pvAAAA//+XQTd/' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/templates/deployment.yaml',
  msg:
   'eJzEUj2P2zAM/SuElhuK5HCr1ivQrTWK4iYBBSMTF6EyJUi00eCq/17I9vnjkgydyiUByfce35MVRvdCKbvAGui3ENe/+XF4OpHgk+FfjlsNnyn6cOmIxXBHgi0KasMAjB1pMOrtDY7PZ0xy/IodQSmHdoWouunxRD6PIABbVzU8XMPWznwW/IFE0aMlMOqTUWDUT6OglAfDOZIdGeuGs5g1VPwL+p7ycW4+h54FSql7Ql30KDSfsbNSa3dkLYzxtr1MnqyENJmbangP0ijbp7RYB1gOHc0HFnRMaRU63A9yK+A6fF22Zpdjr3oN2UlIFyhFX40FX29RNb33TfDOXnbBTZi4DOfwpoohyTahw+qnCfVRN0SZ0uAsvf/W+Y7Lu4GYcm5SONGGE+AsEr+Q7HoAEeWs4fFD819VE2Hr/odsDn2ytA2PeNhFOX0FzbfvP7ZiQxX48Ox31JRhw+pvAAAA//+XQTd/' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/templates/service.yaml',
  msg:
   'eJxsj0FLBDEMhf/Ko1fZAa+9ehdB2FNAYjewg51OaOOCrPPfJbsRQeZUaN738iWxzkfpY15bxuWR2sfcThmv0i9zEWqLGJ/YOFMDGi+SQel6BaanM3ebnnkRbNthBJA8V/ld6rghQPFcUP+gv59QwDe6aOUioPRACZTeKGHbvHeolFunfalkOHzk+iljiuWTDzzcAF273Q0OoX0207uRz3b5eF/WblEzpEqxtcctrLp/yW/OPdNPAAAA//9IVXQO' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/templates/service.yaml',
  msg:
   'eJxsj0FLBDEMhf/Ko1fZAa+9ehdB2FNAYjewg51OaOOCrPPfJbsRQeZUaN738iWxzkfpY15bxuWR2sfcThmv0i9zEWqLGJ/YOFMDGi+SQel6BaanM3ebnnkRbNthBJA8V/ld6rghQPFcUP+gv59QwDe6aOUioPRACZTeKGHbvHeolFunfalkOHzk+iljiuWTDzzcAF273Q0OoX0207uRz3b5eF/WblEzpEqxtcctrLp/yW/OPdNPAAAA//9IVXQO' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/values.yaml',
  msg:
   'eJxkjjFPAzEMRv+K1c6c7piqbBVFLBR1YEG6xU19xVISh9g5uH+P0usE47OeP73NFg40YQ0GM4ZKCpMUiOyL+MARjV7k+QdjDtSNaQvvn6zACggf++PrwyQlohldYOK7cSAfsBDMWBjPgRRM4EyQUZUuwMkEFqkFjGIOaKTdmMZUKAf2+CQ1mYOhnTjildyYAAplUTYpiwO9xbSr4dXB3HdD1zfMNYSTBPaLg334xkXbiFKZ2a8zCSM5eJPL+r7kO52kWLvc3YYOdn3fr2EqtXjS24TP1cHQ97FBpHhLGh53R147vyqpreof+Z+++Q0AAP//24iCrQ==' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: 'chart/gotemplate/values.yaml',
  msg:
   'eJxkjjFPAzEMRv+K1c6c7piqbBVFLBR1YEG6xU19xVISh9g5uH+P0usE47OeP73NFg40YQ0GM4ZKCpMUiOyL+MARjV7k+QdjDtSNaQvvn6zACggf++PrwyQlohldYOK7cSAfsBDMWBjPgRRM4EyQUZUuwMkEFqkFjGIOaKTdmMZUKAf2+CQ1mYOhnTjildyYAAplUTYpiwO9xbSr4dXB3HdD1zfMNYSTBPaLg334xkXbiFKZ2a8zCSM5eJPL+r7kO52kWLvc3YYOdn3fr2EqtXjS24TP1cHQ97FBpHhLGh53R147vyqpreof+Z+++Q0AAP//24iCrQ==' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload called with 
 req.body { isDirectory: false,
  mode: 420,
  path: 'main.go',
  msg:
   'eJyEkb9OxDAMxl/FZOqhqr0R3YYQcAMD4gYWL1HrFovUDqnLCSHeHSXtsbK51ffn9yku+u7djwSTZ0FB4SlqMqhQ0NCxolsvIWvfzOLle5gsn7vsaVt4oTmq9GAKnQ8BVICkj8piKMMiHaRVUZ0hxzSrYabXxEaphgTX2/+PhWbbwXfuYW2K4GSJZazONaA7UggKQ9IJvnRJ8Kgw+ykGAh9j4M4bq0BaRFhGYIH7LnCcCe60pzNLf4UuY9swWfOcWGyo0OX9zaiHHOdj/KPfuGdClOL7yYvLomypNtDCfvTSB3pYpKvQtejqy+j/6wLPRgVYBcoL3Oz3+0vnlv9URLfSnyh9UoXuUESuBuGwornfAAAA//9iDaTs' }
[26/11/19 23:20:59 server.js] [TRACE] responded to PUT /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload with status 200 and body { isDirectory: false,
  mode: 420,
  path: 'main.go',
  msg:
   'eJyEkb9OxDAMxl/FZOqhqr0R3YYQcAMD4gYWL1HrFovUDqnLCSHeHSXtsbK51ffn9yku+u7djwSTZ0FB4SlqMqhQ0NCxolsvIWvfzOLle5gsn7vsaVt4oTmq9GAKnQ8BVICkj8piKMMiHaRVUZ0hxzSrYabXxEaphgTX2/+PhWbbwXfuYW2K4GSJZazONaA7UggKQ9IJvnRJ8Kgw+ykGAh9j4M4bq0BaRFhGYIH7LnCcCe60pzNLf4UuY9swWfOcWGyo0OX9zaiHHOdj/KPfuGdClOL7yYvLomypNtDCfvTSB3pYpKvQtejqy+j/6wLPRgVYBcoL3Oz3+0vnlv9URLfSnyh9UoXuUESuBuGwornfAAAA//9iDaTs' }
[26/11/19 23:20:59 server.js] [TRACE] Checking origin host  matches expected host codewind-workspace0nkmcnl7w7ms2qv8:9191
[26/11/19 23:20:59 server.js] [TRACE] POST /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload/end called with 
 req.body { fileList:
   [ '.cw-settings',
     'Dockerfile',
     'README.md',
     'chart/gotemplate/Chart.yaml',
     'chart/gotemplate/templates/deployment.yaml',
     'chart/gotemplate/templates/service.yaml',
     'chart/gotemplate/values.yaml',
     'main.go' ],
  modifiedList:
   [ '.cw-settings',
     'Dockerfile',
     'README.md',
     'chart/gotemplate/Chart.yaml',
     'chart/gotemplate/templates/deployment.yaml',
     'chart/gotemplate/templates/service.yaml',
     'chart/gotemplate/values.yaml',
     'main.go' ],
  timeStamp: 1574810360292 }
[26/11/19 23:20:59 /portal/routes/projects/remoteBind.route.js] [INFO] Removing locally deleted files from project: gotest123, ID: 2c5df450-10a3-11ea-b04c-b9c7c95ab668 - 
[26/11/19 23:20:59 sharedFunctions.js] [DEBUG] copyProject fromPath: /codewind-workspace/cw-temp/gotest123, toPath: /codewind-workspace/gotest123
[26/11/19 23:20:59 server.js] [TRACE] responded to POST /api/v1/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/upload/end with status 200 and body { fileList:
   [ '.cw-settings',
     'Dockerfile',
     'README.md',
     'chart/gotemplate/Chart.yaml',
     'chart/gotemplate/templates/deployment.yaml',
     'chart/gotemplate/templates/service.yaml',
     'chart/gotemplate/values.yaml',
     'main.go' ],
  modifiedList:
   [ '.cw-settings',
     'Dockerfile',
     'README.md',
     'chart/gotemplate/Chart.yaml',
     'chart/gotemplate/templates/deployment.yaml',
     'chart/gotemplate/templates/service.yaml',
     'chart/gotemplate/values.yaml',
     'main.go' ],
  timeStamp: 1574810360292 }
[26/11/19 23:20:59 User.js] [DEBUG] Project 2c5df450-10a3-11ea-b04c-b9c7c95ab668 file change list received on 1574810360292. chunk number 1 with total chunk 1 for this timestamp.
[26/11/19 23:20:59 FileWatcher.js] [INFO] 2c5df450-10a3-11ea-b04c-b9c7c95ab668 update start time: 1574810459472
[26/11/19 23:20:59 /portal/routes/projects/remoteBind.route.js] [INFO] Total time to sync project gotest123 to build container is 0.416 seconds
[26/11/19 23:20:59 gotest123] [INFO] Extension absolute path is: undefined
[26/11/19 23:20:59 gotest123] [INFO] Project 2c5df450-10a3-11ea-b04c-b9c7c95ab668 file changed
[26/11/19 23:20:59 gotest123] [INFO] cw-settings file changed.
[26/11/19 23:20:59 gotest123] [INFO] Building container for docker project /codewind-workspace/gotest123
[26/11/19 23:20:59 gotest123] [INFO] Extension absolute path is: undefined
[26/11/19 23:20:59 FileWatcher.js] [DEBUG] Received filewatcher module response: 
[26/11/19 23:20:59 FileWatcher.js] [DEBUG] {"statusCode":202}
[26/11/19 23:20:59 gotest123] [INFO] The log name for 2c5df450-10a3-11ea-b04c-b9c7c95ab668 is cw-2c5df450-10a3-11ea-b04c-b9c7c95ab668-6de2b08f8f0c5
[26/11/19 23:20:59 gotest123] [INFO] Container name: cw-gotest123-2c5df450-10a3-11ea-b04c
[26/11/19 23:20:59 gotest123] [INFO] Log name: cw-2c5df450-10a3-11ea-b04c-b9c7c95ab668-6de2b08f8f0c5
[26/11/19 23:20:59 gotest123] [INFO] In K8 environment: true
[26/11/19 23:20:59 gotest123] [INFO] Beginning container build and run stage
[26/11/19 23:20:59 Turbine] [INFO] workspaceSettingsInfoCache cache: {"deploymentRegistry":"docker.io/yangcao77","watcherChunkTimeout":""}
[26/11/19 23:20:59 Turbine] [INFO] workspaceSettingsInfo {"deploymentRegistry":"docker.io/yangcao77","watcherChunkTimeout":""}
[26/11/19 23:20:59 FileWatcher.js] [DEBUG] Portal's event listener: projectSettingsChanged
[26/11/19 23:20:59 FileWatcher.js] [DEBUG] Portal's event listener: {"operationId":"53e1c4ea9b83fc33ee7dac92cdd327b7","projectID":"2c5df450-10a3-11ea-b04c-b9c7c95ab668","contextRoot":"/abc","status":"success"}
[26/11/19 23:20:59 FileWatcher.js] [DEBUG] projectSettingsChanged: success (project 2c5df450-10a3-11ea-b04c-b9c7c95ab668)
[26/11/19 23:20:59 FileWatcher.js] [TRACE] projectSettingsChanged: success (project: {
  "operationId": "53e1c4ea9b83fc33ee7dac92cdd327b7",
  "projectID": "2c5df450-10a3-11ea-b04c-b9c7c95ab668",
  "contextRoot": "/abc",
  "status": "success"
})
[26/11/19 23:20:59 FileWatcher.js] [DEBUG] projectSettingsChanged: success (project 2c5df450-10a3-11ea-b04c-b9c7c95ab668)
[26/11/19 23:20:59 FileWatcher.js] [TRACE] projectSettingsChanged: success (project: {
  "operationId": "53e1c4ea9b83fc33ee7dac92cdd327b7",
  "projectID": "2c5df450-10a3-11ea-b04c-b9c7c95ab668",
  "contextRoot": "/abc",
  "status": "success"
})
[26/11/19 23:20:59 gotest123] [INFO] Extension absolute path is: undefined
[26/11/19 23:20:59 Turbine] [TRACE] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: {"/file-watcher/fwdata/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/2c5df450-10a3-11ea-b04c-b9c7c95ab668.json":"{\"projectID\":\"2c5df450-10a3-11ea-b04c-b9c7c95ab668\",\"projectType\":\"docker\",\"location\":\"/codewind-workspace/gotest123\",\"autoBuildEnabled\":true,\"startMode\":\"run\",\"appPorts\":[],\"language\":\"go\",\"ignoredPaths\":[\"/.project\",\"*/node_modules*\",\"*/.git/*\",\"*/.DS_Store\",\"*/*.swp\",\"*/*.swx\",\"*/4913\",\"/load-test*\",\"*/.dockerignore\",\"*/.gitignore\",\"*/*~\",\"/.settings\"],\"isHttps\":false,\"sentProjectInfo\":false,\"deploymentRegistry\":\"docker.io/yangcao77\",\"appBaseURL\":\"http://cw-gotest123-2c5df450-10a3-11ea-b-che-che731.9.42.26.107.nip.io\"}"}
[26/11/19 23:20:59 Turbine] [INFO] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: projectInfo.deploymentRegistry: docker.io/yangcao77
[26/11/19 23:20:59 Turbine] [INFO] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: Deployment Registry: docker.io/yangcao77
[26/11/19 23:20:59 Turbine] [TRACE] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: {"/file-watcher/fwdata/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/2c5df450-10a3-11ea-b04c-b9c7c95ab668.json":"{\"projectID\":\"2c5df450-10a3-11ea-b04c-b9c7c95ab668\",\"projectType\":\"docker\",\"location\":\"/codewind-workspace/gotest123\",\"autoBuildEnabled\":true,\"startMode\":\"run\",\"appPorts\":[],\"language\":\"go\",\"ignoredPaths\":[\"/.project\",\"*/node_modules*\",\"*/.git/*\",\"*/.DS_Store\",\"*/*.swp\",\"*/*.swx\",\"*/4913\",\"/load-test*\",\"*/.dockerignore\",\"*/.gitignore\",\"*/*~\",\"/.settings\"],\"isHttps\":false,\"sentProjectInfo\":false,\"deploymentRegistry\":\"docker.io/yangcao77\",\"appBaseURL\":\"http://cw-gotest123-2c5df450-10a3-11ea-b-che-che731.9.42.26.107.nip.io\"}"}
[26/11/19 23:20:59 Turbine] [TRACE] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: {"/file-watcher/fwdata/projects/2c5df450-10a3-11ea-b04c-b9c7c95ab668/2c5df450-10a3-11ea-b04c-b9c7c95ab668.json":"{\"projectID\":\"2c5df450-10a3-11ea-b04c-b9c7c95ab668\",\"projectType\":\"docker\",\"location\":\"/codewind-workspace/gotest123\",\"autoBuildEnabled\":true,\"startMode\":\"run\",\"appPorts\":[],\"language\":\"go\",\"ignoredPaths\":[\"/.project\",\"*/node_modules*\",\"*/.git/*\",\"*/.DS_Store\",\"*/*.swp\",\"*/*.swx\",\"*/4913\",\"/load-test*\",\"*/.dockerignore\",\"*/.gitignore\",\"*/*~\",\"/.settings\"],\"isHttps\":false,\"sentProjectInfo\":false,\"deploymentRegistry\":\"docker.io/yangcao77\",\"appBaseURL\":\"http://cw-gotest123-2c5df450-10a3-11ea-b-che-che731.9.42.26.107.nip.io\"}"}
[26/11/19 23:20:59 Turbine] [TRACE] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: Project settings invoked for project 
[26/11/19 23:20:59 Turbine] [INFO] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: Detected other file changes, Codewind will build the project
[26/11/19 23:20:59 Turbine] [TRACE] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: Project settings are {"contextRoot":"abc","internalPort":"","healthCheck":"","isHttps":false,"ignoredPaths":["/.project","*/node_modules*","*/.git/*","*/.DS_Store","*/*.swp","*/*.swx","*/4913","/load-test*","*/.dockerignore","*/.gitignore","*/*~","/.settings"]}
[26/11/19 23:20:59 Turbine] [INFO] 2c5df450-10a3-11ea-b04c-b9c7c95ab668: File change detected. Project will re-build.

[Log truncated, rest looks like .cw-settings change and project build]
```

### Theia container cwctl build version
```
bash-4.4$ pwd
/tmp/vscode-unpacked/eclipse.codewind-plugin.latest.uduoeyeloe.codewind-theia.vsix/extension/bin
bash-4.4$ ls -la
total 20
drwxr-xr-x    3 theia    root           140 Nov 26 23:06 .
drwxr-xr-x    7 theia    root           170 Nov 26 23:06 ..
-rw-r--r--    1 theia    root            73 Nov 26 19:48 .gitignore
-rw-r--r--    1 theia    root           385 Nov 26 19:48 README.txt
-rw-r--r--    1 theia    root             6 Nov 26 19:48 appsody_version.txt
-rw-r--r--    1 theia    root           306 Nov 26 19:48 cli_version.properties
-rw-r--r--    1 theia    root           135 Nov 26 19:48 copy-strings-win.cmd
drwxr-xr-x    2 theia    root            34 Nov 26 23:06 linux
bash-4.4$ cat cli_version.properties 
# Build date: 2019-11-26-19:45:58
build_info.url=https://download.eclipse.org/codewind/codewind-installer/master/18
build_info.linux.SHA-1=6c3a38f752468111176bbcd1a5403142b2f53307
build_info.macos.SHA-1=67235595843d32e31a71a3f919cc1d69ddaa95d0
build_info.win.SHA-1=c13c0af1276b14b3d555ffb43f103da5e06f791d
bash-4.4$ 
```
And at the time of writing this post, I made sure the cwctl build version was the same for the latest devfile:
![Screen Shot 2019-11-26 at 6 33 44 PM](https://user-images.githubusercontent.com/31771087/69681215-646b5900-107b-11ea-8b69-455dd70b84b4.png)

**Comment by tobespc** [comment]
I don't believe this is still an issue.  I suspect the confusion initially with some scenarios working and others not was due to having different version of cwctl left lying around.  I've retested today and cannot reproduce

**Comment by malincoln** [comment]
@yangcao77 can you confirm whether this is still an issue or not?  Thanks

**Comment by yangcao77** [comment]
I'm still able to reproduce the issue on my cluster with latest master image. 
Like @maysunfaisal and I mentioned, `cwctl` is behaving differently on different environments. I can always reproduce this even with a new install of che. 

I can provide my cluster for debug purpose.

**Comment by yangcao77** [comment]
Correct myself, I'm not seeing this issue anymore. 
Instead, I saw another issue that `projectChanged` event got emitted right after `projectCreation` event. Will investigate more on that one. 
Closing this issue
